ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncDownloaderTest.java

```diff
@@ -413,17 +401,7 @@ public void shouldResetFastSyncStateAndRestartProcessIfANonFastSyncExceptionOccu
     // A real chain downloader would cause the chainFuture to complete when cancel is called.
     chainFuture.completeExceptionally(new CancellationException());
 
-    verify(fastSyncActions, times(2)).waitForSuitablePeers(FastSyncState.EMPTY_SYNC_STATE);
-    verify(fastSyncActions, times(2)).selectPivotBlock(FastSyncState.EMPTY_SYNC_STATE);
-    verify(fastSyncActions).downloadPivotBlockHeader(secondSelectPivotBlockState);
-    verify(storage).storeState(secondDownloadPivotBlockHeaderState);
-    verify(fastSyncActions).createChainDownloader(secondDownloadPivotBlockHeaderState);
-    verify(worldStateDownloader).run(secondPivotBlockHeader);
-    verifyNoMoreInteractions(fastSyncActions, worldStateDownloader, storage);
-
-    secondWorldStateFuture.complete(null);
-
-    assertThat(result).isCompletedWithValue(secondDownloadPivotBlockHeaderState);
+    verify(fastSyncActions).scheduleFutureTask(any(Supplier.class), any(Duration.class));
```

Comment:
I think you should be able to validate the second run if you use `thenAnswer` to go ahead and run the future supplier with something like: 
```
when(fastSyncActions).scheduleFutureTask(any(), any())
    .thenAnswer(invocation -> invocation.getArgument(1).get());
```

Reply:
Done